### ISSUE TITLE
Camera Resolution? Dark Images with AVCaptureSessionPresetPhoto

### ISSUE THREAD

**Issue by rocket-enterprise** [AlexLittlejohn/ALCameraViewController#72]
Hello,

thanks for sharing that great plugin! I have some issues when ill try to set the quality to AVCaptureSessionPresetPhoto. 

The result is, that every image is totally dark. When ill press the capture button, ill see (for maybe 0.5 seconds) the correct, good looking bright image, and then its getting quite dark. Also in the Media Library from the Pad/Phone is the image quite dark.

When ill set to

AVCaptureSessionPresetHigh

Everything is fine, and the image is shown with much more brightness. Any ideas what could be a problem? ill just know that the:

AVCaptureSessionPresetHigh is normally used to make Video, and AVCaptureSessionPresetPhoto is for Photos. It looks like that there is any camera filter applied.

Any ideas? Thanks in advance.

**Comment by AlexLittlejohn** [comment]
I have take a look at this and it seems it is caused by the `ContinuousAutoExposure` setting. I'm investigating a solution.

**Comment by rocket-enterprise** [comment]
Thanks John for your quick reply. I will have a look at that point!

**Comment by rocket-enterprise** [comment]
Hello AlexLittleJohn,

I am still trying to find any fix for that problem. Can u give me little more information what you think what cause the problem? The ContinuousAutoExposure should be only active, when user tries to set a focus point, or?

Because that function here:

 public func focusCamera(toPoint: CGPoint) -> Bool {

Should be normally not beeing called when ill simple make an image.

If i set the ExposureMode directly in CapurePhoto, nothing changes. Its stange when you see for about 0,5 seconds the correct image, and then its getting very very dark und unusable.

**Comment by AlexLittlejohn** [comment]
After some research it appears that this occurs if we take photos too quickly after instantiating the camera. The camera input seems to need some time to open the shutters and adjust the white balance and so on. The solution appears to be putting a artificial delay on capturing the photo after pressing the button.

**Comment by rocket-enterprise** [comment]
Thanks again for your quick response!
When ill try to get the value of adjustingExposure its always false) 
Where should i set the camera settings? Ill try at the capturePhoto() function:

    internal func capturePhoto() {
        guard let output = cameraView.imageOutput, connection = output.connectionWithMediaType(AVMediaTypeVideo) else {
            return
        }
        cameraView.session.sessionPreset = AVCaptureSessionPresetPhoto
        print(cameraView.device.adjustingExposure) // always false
        
Whats really strange is, that for one short moment the image looks fine. (also in Preview) - it looks like that something applies an image filter to to final image (and it looses 80% brightness...)

Would be really great to find an solution here, and i´d like to donate some money for sharing that plugin!

**Comment by AlexLittlejohn** [comment]
There are a few adjustment properties to listen to namely 'device.adjustingFocus)', 'device.adjustingExposure' and 'device.adjustingWhiteBalance'.

**Comment by AlexLittlejohn** [comment]
Adding a while loop which checks for any of those being true solved this for me in my admittedly limited testing.

**Comment by rocket-enterprise** [comment]
Where do you check it? In the capturePhoto() function?

**Comment by rocket-enterprise** [comment]
Just print it out results in always false for any value:

    internal func capturePhoto() {
        guard let output = cameraView.imageOutput, connection = output.connectionWithMediaType(AVMediaTypeVideo) else {
            return
        }
        
        cameraView.session.sessionPreset = AVCaptureSessionPresetPhoto
        
        print(cameraView.device.adjustingExposure)
        print(cameraView.device.adjustingWhiteBalance)
        print(cameraView.device.adjustingFocus)

**Comment by AlexLittlejohn** [comment]
```
    public func capturePhoto(completion: CameraShotCompletion) {
        dispatch_async(cameraQueue) {
            
            if let device = self.device {
                while device.adjustingWhiteBalance || device.adjustingExposure || device.adjustingFocus {
                    print("waiting")
                }
            }
            
            let orientation = AVCaptureVideoOrientation(rawValue: UIDevice.currentDevice().orientation.rawValue)!
            takePhoto(self.imageOutput, videoOrientation: orientation, cropSize: self.frame.size) { image in
                dispatch_async(dispatch_get_main_queue()) {
                    completion(image)
                }
            }
        }
    }
```

**Comment by rocket-enterprise** [comment]
You are right - sometime it works (takes up to 2 seconds to make an image) - and works in 1/5 times with an good result (rest is totally dark like before).

How does apple do this with there default app? There everything works fine without waiting 2 seconds for 1 simple image. And could that be really such a quite difference between the 1920x1080 resolution and that one with 34xx x 22xx? When ill only set to High Resoltion all images are fine like in the apple default camera app.

**Comment by rocket-enterprise** [comment]
Ill found out that the problem do not exist on the iPad Pro 9.7". But with all other iPads still really dark and unuseable.
